test/test_specs.py

```diff
@@ -1409,6 +1433,21 @@ def test_non_tensor(self):
             == NonTensor((2, 3, 4), device="cpu")
         )
 
+    @pytest.mark.parametrize("input_type", ["spec", "nontensor", "nontensorstack"])
+    def test_choice(self, input_type):
+        if input_type == "spec":
+            stack = torch.stack([Bounded(0, 2.5, ()), Bounded(10, 12, ())])
+        elif input_type == "nontensor":
+            stack = torch.stack([NonTensorData("a"), NonTensorData("b")])
+        elif input_type == "nontensorstack":
+            stack = torch.stack(
+                [NonTensorStack("a", "b", "c"), NonTensorStack("d", "e", "f")]
+            )
+
+        spec = Choice(stack)
+        with pytest.raises(NotImplementedError):
```

Comment:
is this intentional? Why can't we just expand the sub-specs?
I would have thought that this would hold
```python
choice = Choice(torch.stack([Bounded(0, 2.5, (3,)), Bounded(10, 12, (3,))]))
assert choice.shape == (3,)
choice_expand = choice.expand(10, 3)
assert choice_expand.shape == (10, 3,)
assert choice_expand.rand().shape == (10, 3)
```

Reply:
Fixed